app/src/main/java/com/sun/heartrate/ui/heartbeat/HeartbeatFragment.kt

```diff
@@ -81,18 +91,66 @@ class HeartbeatFragment : Fragment(), HeartbeatContract.View, View.OnClickListen
         }
     }
     
-    private fun upDateIuButton(isCheckCamera: Boolean) {
-        if (isCheckCamera) {
-            buttonStart.text = getText(R.string.label_pause)
+    private fun updateUiScreenHeartbeat() {
+        updateUiButton(heartbeatPresenter.isRecording())
+        displayHeatRate(0)
+        displayGuideline(R.string.title_measuring_heart_rate)
+        displayImageViewHeart(heartbeatPresenter.isRecording())
+        updateUiProgressBar(heartbeatPresenter.isRecording())
+    }
+    
+    private fun updateUiButton(isRecording: Boolean) {
```

Comment:
Giảm code lặp nhau. Và nên đặt tên method trừu tượng hơn là máy móc "update UI".
```suggestion
    private fun showButtonLabel(isRecording: Boolean) {
        val stringLabelId = if(isRecording) R.string.label_pause else R.string.label_start
        buttonStart?.text = getString(stringLabelId)
    }
```